front/components/agent_builder/capabilities/mcp/utils/sheetUtils.ts

```diff
@@ -1,6 +1,243 @@
+import type { UseFormReturn } from "react-hook-form";
+
+import type { MCPFormData } from "@app/components/agent_builder/AgentBuilderFormContext";
+import type {
+  SelectedTool,
+  SheetMode,
+} from "@app/components/agent_builder/capabilities/mcp/MCPServerViewsSheet";
+import type {
+  AgentBuilderAction,
+  ConfigurationPagePageId,
+} from "@app/components/agent_builder/types";
+import { TOOLS_SHEET_PAGE_IDS } from "@app/components/agent_builder/types";
+import { getMcpServerViewDisplayName } from "@app/lib/actions/mcp_helper";
+import type { MCPServerViewType } from "@app/lib/api/mcp";
+import { pluralize } from "@app/types";
+
 export function isValidPage<T extends Record<string, string>>(
   pageId: string,
   pageIds: T
 ): pageId is T[keyof T] {
   return Object.values(pageIds).includes(pageId);
 }
+
+export function getInitialPageId(
+  mode: SheetMode | null
+): ConfigurationPagePageId {
+  const currentMode = mode?.type ?? "add";
+
+  if (currentMode === "edit" || currentMode === "configure") {
+    return TOOLS_SHEET_PAGE_IDS.CONFIGURATION;
+  }
+  if (currentMode === "info") {
+    return TOOLS_SHEET_PAGE_IDS.INFO;
+  }
+  return TOOLS_SHEET_PAGE_IDS.TOOL_SELECTION;
+}
+
+export function getInitialConfigurationTool(
+  mode: SheetMode | null
+): AgentBuilderAction | null {
+  if (!mode) {
+    return null;
+  }
+
+  if (mode.type === "edit" || mode.type === "configure") {
+    return mode.action;
+  }
+
+  return null;
+}
+
+export interface FooterButtonOptions {
+  currentPageId: ConfigurationPagePageId;
+  modeType: SheetMode["type"];
+  selectedToolsInSheet: SelectedTool[];
+  form: UseFormReturn<MCPFormData>;
+  onCancel: () => void;
+  onModeChange: (mode: SheetMode | null) => void;
+  onAddSelectedTools: () => void;
+  onConfigurationSave: (data: MCPFormData) => void;
+  resetToSelection: () => void;
+}
+
+export function getFooterButtons({
```

Comment:
You might missing a case because I cannot validate tools 🤔
<img width="600" height="864" alt="Image" src="https://github.com/user-attachments/assets/108249e8-1b83-47b8-a30f-9e816b90df82" />